File: app/src/main/java/me/aluceps/sandbox/view/main/MainContract.java
package me.aluceps.sandbox.view.main;

import java.util.List;

import me.aluceps.sandbox.model.ConnpassEvent;
import me.aluceps.sandbox.view.BasePresenter;
import me.aluceps.sandbox.view.BaseView;

public class MainContract {
    interface View extends BaseView {

        void initializeRecyclerView();

        MainAdapter getAdapter();

        void setEvents(List<ConnpassEvent.Event> events);

        void clear();

        void showProgressBar();

        void hideProgressBar();
    }

    interface Presenter<T extends BaseView> extends BasePresenter<T> {

        void load();

        boolean isLoading();

        void refresh();

        void destroy();
    }
}

File: app/src/main/java/me/aluceps/sandbox/view/main/MainContract.kt
package me.aluceps.sandbox.view.main

import me.aluceps.sandbox.model.ConnpassEvent
import me.aluceps.sandbox.view.BasePresenter
import me.aluceps.sandbox.view.BaseView

class MainContract {
    interface View : BaseView {

        fun initializeRecyclerView()

        fun setData(data: List<ConnpassEvent.Event>)

        fun clear()

        fun showProgressBar()

        fun hideProgressBar()
    }

    interface Presenter<T : BaseView> : BasePresenter<T> {

        fun isLoading(): Boolean

        fun load()

        fun refresh()

        fun destroy()
    }
}
